bd-client-stats/src/stats.rs

```diff
@@ -87,6 +87,15 @@ pub trait SerializedFileSystem: Sync {
     // For now we just read the entire thing and then decompress it in memory. We can consider
     // streaming later.
     let compressed_bytes = self.read_file(path).await?;
+
+    // We should never write empty files. If there is no data this was a partial write, full disk
+    // issue, or some other problem.
+    // TODO(mattklein123): Compression gives some protection against further corruption but we
```

Comment:
It would probably be worth doing some generic protobuf file writing logic that always handles the crc header, as I'm sure our config persistence is also susceptible to file corruption

Reply:
Yes agreed. I can look at this as a follow up.